### ISSUE TITLE
Clarify a way to access host attribute with `inject` function API

### ISSUE THREAD

**Issue by wszydlak** [angular/angular#53648]
### Which @angular/* package(s) are relevant/related to the feature request?

core

### Description

This is more question than feature request. Is it possible to access host attribute using `inject` function?

Current way is to use `Attribute` decorator in constructor parameter:

```ts
class TestDirective {
  constructor(@Attribute('token') token: string) {}
}
```

I've seen a solution with using `inject` function:

```ts
class TestDirective {
  token: string = inject(new Attribute('token') as any);
}
```

However this requires to cast `Attribute` type to `any` to allow passing it as parameter to `inject` function.

Maybe we should cast to `Type<string>`?:

```ts
class TestDirective {
  token = inject(new Attribute('token') as unknown as Type<string>);
}
```

Or maybe we should avoid using `Attribute` and inject `ElementRef` instead?:

```ts
class TestDirective {
  token = inject(ElementRef<HTMLElement>).nativeElement.getAttribute('token');
}
```

### Proposed solution

There could be an official information if we should use `inject` or keep using `Attribute` decorator in `constructor`

### Alternatives considered

??

**Comment by JeanMeche** [comment]
Current PR is #53672

**Comment by crisbeto** [comment]
Closing as fixed by #53672.
